bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/breadcrumb/BreadcrumbImpl.java

```diff
@@ -256,10 +259,13 @@ private boolean isCategoryPage(Page page) {
 
     private boolean isSpecificPage(Page page) {
         // The product or category page might be in a Launch so we first extract the paths of the production versions
-        String productPagePath = productPage.getPath().substring(productPage.getPath().lastIndexOf("/content/"));
-        String categoryPagePath = categoryPage.getPath().substring(categoryPage.getPath().lastIndexOf("/content/"));
+        String productPagePath = productPage == null ? null
+            : productPage.getPath().substring(productPage.getPath().lastIndexOf("/content/"));
+        String categoryPagePath = categoryPage == null ? null
+            : categoryPage.getPath().substring(categoryPage.getPath().lastIndexOf("/content/"));
 
         String path = page.getPath();
-        return (path.contains(productPagePath + "/") || path.contains(categoryPagePath + "/"));
+        return (productPagePath != null && path.contains(productPagePath + "/") ||
```

Comment:
trivial: while we are on it this check should probably be done using `Text.isDescendant` https://www.adobe.io/experience-manager/reference-materials/6-5/javadoc/com/day/text/Text.html#isDescendantOrEqual-java.lang.String-java.lang.String-

Reply:
Also https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#contains-java.lang.CharSequence-java.lang.CharSequence (which is already used in this class) can be used that includes the null check.